lib/form/js/repeatable.js

```diff
@@ -0,0 +1,346 @@
+            if (name) {
+              name = splitPop(name, '--');
+              child.setAttribute('name', name);
+            }
+          }
+          else if (child.htmlFor && child.htmlFor !== '') {
+            child.htmlFor = splitPop(child.htmlFor, '--');
+          }
+        });
+      },
+
+      // Handles delete events
+      deleteHandler = function deleteHandler(e) {
+        var node = e.currentTarget.parentNode,
+            siblingNode = node.nextElementSibling,
+            container,
+            parent = node.parentNode,
+            attribute = parent.id;
+
+        // removes child and update length
+
+        parent.removeChild(node);
+        allRepeatables[attribute].length--;
+        deleteIdsAndInputs(node.querySelectorAll('*'));
+
+        // change ids and names of subsequest instances
+        while (siblingNode) {
+          if (siblingNode.getAttribute('class') === 'form--repeatable') {
+            updateInstance(siblingNode);
+          }
+          siblingNode = siblingNode.nextElementSibling;
+        }
+
+        // remove index if there is just one instance left
+        if (allRepeatables[attribute].length === 1) {
+          container = parent.querySelector('.form--repeatable');
+          removeIndex(container);
+          container.removeChild(container.querySelector('.form--delete'));
+        }
+      },
+
+      // Add new input to allInputs
+      setIDsAndInputs = function setIDsAndInputs(idMap) {
+        Object.keys(idMap).forEach(function setIdandInput(id) {
+          if (allInputs.hasOwnProperty(id)) {
+            allInputs[idMap[id]] = Object.assign({}, allInputs[id]);
+            if (!allIDs.hasOwnProperty(idMap[id])) {
+              allIDs[idMap[id]] = document.getElementById(idMap[id]);
+
+              // Adds validation to new input
+              allIDs[idMap[id]].addEventListener(allInputs[idMap[id]].validation.on, validate);
+            }
+          }
+        });
+      },
+
+      // Create and return delete button
+      newDeleteButton = function newDeleteButton(attribute) {
+        var element = document.createElement('button');
+        element.type = 'button';
+        element.textContent = 'Delete';
+        element.id = attribute + '--delete';
+        element.setAttribute('class', 'form--delete');
+        element.addEventListener('click', deleteHandler);
+
+        return element;
+      },
+
+      // Update values of current instance
+      updateCurrentInstance = function updateCurrentInstance(instance, attribute) {
+        var container = instance,
+            index = allRepeatables[attribute].length;
+
+        // Adds new delete button to container
+        container.appendChild(newDeleteButton(attribute));
+
+        // Adds index to id
+        container.id = container.id + '--' + (index - 1);
+
+        // Updates all children of container
+        container.querySelectorAll('*').forEach(function setAttributes(node) {
```

Comment:
Did you test this across browsers? [`querySelectorAll`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) returns a [`NodeList`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) which is _not_ an Array and does not have `forEach` available. In browsers that implement `NodeList` properly, this will break.

You should either loop over these with a `for` loop, or use `Array.prototype.forEach.call()` instead. This holds true for all places you do this